src/Rules/Deprecations/DeprecatedHookImplementation.php

```diff
@@ -46,6 +43,36 @@ public function processNode(Node $node, Scope $scope) : array
         $function_name = $node->name->toString();
         $hook_name = substr_replace($function_name, "hook", 0, strlen($module_name));
 
+        // @todo replace this hardcoded logic with something more intelligent and extensible.
+        if ($hook_name === 'hook_field_widget_form_alter') {
+            return [
+                RuleErrorBuilder::message(
+                    "Function $function_name implements $hook_name which is deprecated in drupal:9.2.0 and is removed from drupal:10.0.0. Use hook_field_widget_single_element_form_alter instead.",
+                )->build()
+            ];
+        }
+        if (str_starts_with($hook_name, 'hook_field_widget_') && str_ends_with($hook_name, '_form_alter')) {
+            return [
+                RuleErrorBuilder::message(
+                    "Function $function_name implements $hook_name which is deprecated in drupal:9.2.0 and is removed from drupal:10.0.0. Use hook_field_widget_single_element_WIDGET_TYPE_form_alter instead.",
+                )->build()
+            ];
+        }
+        if ($hook_name === 'hook_field_widget_multivalue_form_alter') {
+            return [
+                RuleErrorBuilder::message(
+                    "Function $function_name implements $hook_name which is deprecated in drupal:9.2.0 and is removed from drupal:10.0.0. Use hook_field_widget_complete_form_alter instead.",
+                )->build()
+            ];
+        }
+        if (str_starts_with($hook_name, 'hook_field_widget_multivalue_') && str_ends_with($hook_name, '_form_alter')) {
+            return [
+                RuleErrorBuilder::message(
+                    "Function $function_name implements $hook_name which is deprecated in drupal:9.2.0 and is removed from drupal:10.0.0. Use hook_field_widget_complete_WIDGET_TYPE_form_alter instead.",
+                )->build()
+            ];
+        }
```

Comment:
This needs a refactor. Ideally, it would be a better format as a static array in a helper method.